### PR TITLE
Switch to using photo picker

### PR DESCRIPTION
**Pull request by kevinMerizalde**
To solve the issue #1261 
I apply the logic that is explained here: https://developer.android.com/training/data-storage/shared/photopicker#java

First, I need to look for situations in which the user is allowed to select an image from their gallery or files directory. I found that the user is allowed to select an image in three main cases:

1. When creating a card and an icon or image can be defined, or in the Photos tab (to define the front and back image).
2. When editing a card and an icon or image can be defined, or in the Photos tab (to define the front and back image).
3. When extracting a barcode or QR code.

After identify these cases I search the code that display the gallery or the file directory and I adapt that code to use the new photo picker approach instead the old one (Only if the Android version of the phone is superior to Android R (11), because if I want to apply this approach to old versions of android, I need to enable the automatic installation of the backported photo picker module through Google Play services and the issue says that is better to not install things without the permission of the user and I agree with that).

Basically, I add some dependencies of androidx and edit the next functions:
* addFromImageAfterPermission() - In the ScanActivity
* selectImageFromGallery(int type) - In the LoyaltyCardEditActivity

If the phone had an Android version oldest than 11 then the old code is executed, if not then the new photo picker is used.

### CODE REVIEW

File: app/src/main/java/protect/card_locker/ScanActivity.java [hunk 1]
```diff
@@ -226,28 +230,41 @@ public void addFromImage(View view) {
...
-        customBarcodeScannerBinding.cardInputContainer.setBackgroundColor(show ? obtainThemeAttribute(com.google.android.material.R.attr.colorSurface) : Color.TRANSPARENT);
-        customBarcodeScannerBinding.cameraPermissionDeniedLayout.getRoot().setVisibility(show ? View.VISIBLE : View.GONE);
-    }
+        private void showCameraPermissionMissingText ( boolean show){
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedClickableArea.setOnClickListener(show ? v -> {
+                navigateToSystemPermissionSetting();
+            } : null);
+            customBarcodeScannerBinding.cardInputContainer.setBackgroundColor(show ? obtainThemeAttribute(com.google.android.material.R.attr.colorSurface) : Color.TRANSPARENT);
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.getRoot().setVisibility(show ? View.VISIBLE : View.GONE);
+        }
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/ScanActivity.java:267]: This spacing change is very inconsistent with the rest of the code, please revert it.

File: app/build.gradle [hunk 2]
```diff
@@ -84,7 +84,18 @@ android {
...
+            because("kotlin-stdlib-jdk7 is now a part of kotlin-stdlib")
+        }
+        implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.8.0") {
+            because("kotlin-stdlib-jdk8 is now a part of kotlin-stdlib")
+        }
+    }
 // AndroidX
+    implementation 'androidx.activity:activity:1.7.1'
+    implementation 'androidx.activity:activity-compose:1.7.1'
+    implementation 'androidx.activity:activity-ktx:1.7.1'
```
Comment (TheLastProject) [inline app/build.gradle:98]: Are these new dependencies necessary? `androidx.activity:activity:1.7.1` is according to the documentation, but I don't see `-compose` and `-ktx` mentioned as necessary.

File: app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java [hunk 3]
```diff
@@ -1135,18 +1143,31 @@ private void takePhotoForCard(int type) {
...
-            mPhotoPickerLauncher.launch(chooserIntent);
-        } catch (ActivityNotFoundException e) {
-            Toast.makeText(getApplicationContext(), R.string.failedLaunchingPhotoPicker, Toast.LENGTH_LONG).show();
-            Log.e(TAG, "No activity found to handle intent", e);
+            try {
+                mPhotoPickerLauncher.launch(chooserIntent);
+            } catch (ActivityNotFoundException e) {
+                Toast.makeText(getApplicationContext(), R.string.failedLaunchingPhotoPicker, Toast.LENGTH_LONG).show();
+                Log.e(TAG, "No activity found to handle intent", e);
+            }
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java:1170]: If this version check necessary? Looking at https://developer.android.com/reference/androidx/activity/result/contract/ActivityResultContracts.PickVisualMedia it states: "@[RequiresApi](https://developer.android.com/reference/androidx/annotation/RequiresApi)(value = 19)" and "[Intent.ACTION_OPEN_DOCUMENT](https://developer.android.com/reference/android/content/Intent.html#ACTION_OPEN_DOCUMENT()) is used as a final fallback on all Android API 19 or higher devices.". Given Catima's minSdk is 21, we should be able to always follow this code path.

File: app/src/main/java/protect/card_locker/ScanActivity.java [hunk 4]
```diff
@@ -226,28 +230,41 @@ public void addFromImage(View view) {
...
+            contentIntent.setType("image/*");
+
+            Intent chooserIntent = Intent.createChooser(photoPickerIntent, getString(R.string.addFromImage));
+            chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[]{contentIntent});
+            try {
+                photoPickerLauncher.launch(chooserIntent);
+            } catch (ActivityNotFoundException e) {
+                Toast.makeText(getApplicationContext(), R.string.failedLaunchingPhotoPicker, Toast.LENGTH_LONG).show();
+                Log.e(TAG, "No activity found to handle intent", e);
+            }
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/ScanActivity.java:257]: See comment above

**Review by TheLastProject — CHANGES_REQUESTED** [review]
Haven't had time to test this yet, but I have several questions. Could you please look at these?

File: app/src/main/java/protect/card_locker/ScanActivity.java (hunk 1 above)
Comment (kevinMerizalde) [reply app/src/main/java/protect/card_locker/ScanActivity.java]: I think the IDE auto format that, I didn't realize that happen. I already change it

File: app/build.gradle (hunk 2 above)
Comment (kevinMerizalde) [reply app/build.gradle:98]: When I click the link of the documentation then appear that 3 libraries but I tried using only  `androidx.activity:activity:1.7.1` and is working too. so let me remove the other ones.

File: app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java (hunk 3 above)
Comment (kevinMerizalde) [reply app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java:1170]: I think is better to check the version and if is lower than Android 11, do it like the old version and in that way make sure the same functionality.

**Comment by TheLastProject** [comment]
I've tested and can tell it works on my Android 13 phone. After testing on an Android 5 emulator I kinda understand your decision to have the two code paths as it somehow opens the file manager on Android 5 instead of the gallery.

I do still prefer just a single code path, also for consistency in behaviour with other apps.

But this needs to be delayed anyway given the Photo Picker currently only shows internal camera images and thus will lack many many images the user my want to select. Maybe it'll get added later.

Thanks for your contribution, it's sad Google hasn't developed the Photo Picker in a way where it'll be useful for most users.
